### PR TITLE
chore(worker-utils): Cleanup WorkerThread code

### PR DESCRIPTION
**Pull request by ibgreen**
Prep PR for adding Node.js worker support

### CODE REVIEW

File: modules/core/src/lib/loader-utils/parse-with-worker.js [hunk 1]
```diff
@@ -83,9 +83,9 @@ async function onWorkerMessage({worker, data, resolve, reject}) {
     case 'parse':
       try {
         const result = await parse(data.arraybuffer, data.options, data.url);
-        worker.postMessage({type: 'parse-done', id: data.id, result}, getTransferList(result));
+        workerThread.postMessage({type: 'parse-done', id: data.id, result}, result);
```
Comment (Pessimistress) [inline modules/core/src/lib/loader-utils/parse-with-worker.js:86]: `postMessage` only accepts one argument

**Review by Pessimistress — APPROVED** [review]

File: modules/core/src/lib/loader-utils/parse-with-worker.js (hunk 1 above)
Comment (ibgreen) [reply modules/core/src/lib/loader-utils/parse-with-worker.js:86]: This is now a method on `WorkerThread` class. The underlying worker will vary between browser and node.

Comment (Pessimistress) [reply modules/core/src/lib/loader-utils/parse-with-worker.js:86]: I assume you are using Node.js' worker module for the node version. It has a similar interface as web worker. I don't see why the second argument is needed.

Comment (Pessimistress) [reply modules/core/src/lib/loader-utils/parse-with-worker.js:86]: Yes I'm talking about `WorkerThread.postMessage`. `Worker.postMessage` takes two arguments, the second of which is generated inside `WorkerThread.postMessage`.

Comment (ibgreen) [reply modules/core/src/lib/loader-utils/parse-with-worker.js:86]: Yeah there was a type hole here, your eagle eyes caught it. This method seems identical but I have noticed a surprising amount of subtle differences between the two Worker classes, so I went with the wrapper approach, we can revisit later if it makes sense.

**Pull request merged**